File: skills/cow-arms-technology-analysis/scripts/R/01_load_and_clean.R
#!/usr/bin/env Rscript
# 01_load_and_clean.R — load COW Arms Tech, filter to a country-year subset, sanity-check.
#
# Codebook: ArmsTechnologyV1/CoW_codebook.pdf §§2–3.1 (Hariri & Wingender 2025).
# Missing values are `.` in Stata, parsed to NA on CSV import — do NOT pass na = "-9".
#
# Usage:
#   export COW_ARMS_DATA_DIR=/path/to/ArmsTechnologyV1
#   Rscript 01_load_and_clean.R
# (Without the env var, falls back to ../../ArmsTechnologyV1 relative to this script.)

suppressPackageStartupMessages({
  library(readr)
  library(dplyr)
})

data_dir <- Sys.getenv("COW_ARMS_DATA_DIR",
                       unset = file.path("..", "..", "..", "..", "ArmsTechnologyV1"))
wide_path <- file.path(data_dir, "cow_arms_tech_wide.csv")

stopifnot(file.exists(wide_path))
wide <- read_csv(wide_path, show_col_types = FALSE)

# Major-power illustration: US (2), Russia/USSR (365), China (710).
panel <- wide |>
  filter(ccode %in% c(2, 365, 710), year >= 1900, year <= 2023) |>
  select(ccode, statename, year, total_use, version)

cat("Rows: ", nrow(panel), "\n")
cat("Missing total_use: ", sum(is.na(panel$total_use)), "\n")
cat("Range of total_use: ",
    paste(range(panel$total_use, na.rm = TRUE), collapse = "–"), "\n\n")

# Per-country summary
panel |>
  group_by(statename) |>
  summarise(n_years = n(),
            n_missing = sum(is.na(total_use)),
            first_year = min(year),
            last_year = max(year),
            mean_total_use = round(mean(total_use, na.rm = TRUE), 2),
            .groups = "drop") |>
  print()

# Save the cleaned subset for downstream scripts.
dir.create("output", showWarnings = FALSE)
write_csv(panel, "output/panel_majors_1900_2023.csv")
cat("\nWrote output/panel_majors_1900_2023.csv\n")

File: skills/cow-arms-technology-analysis/scripts/python/01_load_and_clean.py
#!/usr/bin/env python3
"""01_load_and_clean.py — load COW Arms Tech, filter to a country-year subset.

Codebook: ArmsTechnologyV1/CoW_codebook.pdf §§2-3.1 (Hariri & Wingender 2025).
Missing values are `.` in Stata; pandas parses blank CSV fields to NaN
automatically — do NOT pass na_values=[-9] or na_values=["-9"].

Usage:
    export COW_ARMS_DATA_DIR=/path/to/ArmsTechnologyV1
    python 01_load_and_clean.py
"""

import os
from pathlib import Path
import pandas as pd

data_dir = Path(os.environ.get(
    "COW_ARMS_DATA_DIR",
    Path(__file__).resolve().parents[3] / "ArmsTechnologyV1",
))
wide_path = data_dir / "cow_arms_tech_wide.csv"
assert wide_path.exists(), f"Not found: {wide_path}"

wide = pd.read_csv(wide_path)

# Major-power illustration: US (2), Russia/USSR (365), China (710).
mask = wide["ccode"].isin([2, 365, 710]) & wide["year"].between(1900, 2023)
panel = wide.loc[mask, ["ccode", "statename", "year", "total_use", "version"]].copy()

print(f"Rows: {len(panel)}")
print(f"Missing total_use: {panel['total_use'].isna().sum()}")
print(f"Range of total_use: {panel['total_use'].min()}–{panel['total_use'].max()}")
print()

# Per-country summary
summary = (
    panel.groupby("statename")
    .agg(
        n_years=("year", "size"),
        n_missing=("total_use", lambda s: s.isna().sum()),
        first_year=("year", "min"),
        last_year=("year", "max"),
        mean_total_use=("total_use", lambda s: round(s.mean(), 2)),
    )
    .reset_index()
)
print(summary.to_string(index=False))

Path("output").mkdir(exist_ok=True)
panel.to_csv("output/panel_majors_1900_2023.csv", index=False)
print("\nWrote output/panel_majors_1900_2023.csv")
